File: poribar_health/app/api/v1/public/geo.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy import select
from sqlalchemy.orm import Session

from app.database import get_db
from app.models.geo import District, Division, Upazila, Union
from app.schemas.geo import DistrictOut, DivisionOut, UpazilaOut, UnionOut

router = APIRouter(prefix="/geo", tags=["Geo (Public)"])


@router.get("/divisions", response_model=list[DivisionOut])
def list_divisions(db: Session = Depends(get_db)):
    return db.scalars(select(Division)).all()


@router.get("/districts", response_model=list[DistrictOut])
def list_districts(division_id: int = Query(...), db: Session = Depends(get_db)):
    return db.scalars(select(District).where(District.division_id == division_id)).all()


@router.get("/upazilas", response_model=list[UpazilaOut])
def list_upazilas(district_id: int = Query(...), db: Session = Depends(get_db)):
    return db.scalars(select(Upazila).where(Upazila.district_id == district_id)).all()


@router.get("/unions", response_model=list[UnionOut])
def list_unions(upazila_id: int = Query(...), db: Session = Depends(get_db)):
    return db.scalars(select(Union).where(Union.upazila_id == upazila_id)).all()

File: frontend/src/api/geo.ts
import { api } from "./client";
import type { Division, District, Upazila, Union } from "@/types";

export const geoApi = {
  divisions: () => 
    api.get<Division[]>("/geo/divisions").then((r) => r.data),

  districts: (divisionId: number) => 
    api.get<District[]>("/geo/districts", { params: { division_id: Number(divisionId) } }).then((r) => r.data),

  upazilas: (districtId: number) => 
    api.get<Upazila[]>("/geo/upazilas", { params: { district_id: Number(districtId) } }).then((r) => r.data),

  unions: (upazilaId: number) => 
    api.get<Union[]>("/geo/unions", { params: { upazila_id: Number(upazilaId) } }).then((r) => r.data),
};